test/utils/driver.js

```diff
@@ -83,11 +87,18 @@ const Driver = {
    * @return Promise<Object>
    */
   getReduxState: async function() {
-    await this.setTargetToBrowserWindow();
-
-    return await this.client.execute(() =>
-      JSON.stringify(require('electron').ipcRenderer.store.getState()))
-      .then(res => JSON.parse(res.value));
+    const oldHandle = await this.setTargetToBrowserWindow();
+
+    try {
+      return await this.client.execute(() =>
+        JSON.stringify(require('electron').ipcRenderer.store.getState()))
+        .then(res => JSON.parse(res.value));
+    } finally {
+      // Here's a situation where a decorator would be useful.
```

Comment:
Agreed -- after running a few tests, the overhead of switching contexts isn't much at all. We're slowly piecing together the story for these tests so should be good to abstract this out shortly.